sdv/sampling/independent_sampler.py

```diff
@@ -98,16 +100,18 @@ def _finalize(self, sampled_data):
         final_data = {}
         for table_name, table_rows in sampled_data.items():
             synthesizer = self._table_synthesizers.get(table_name)
-            metadata = synthesizer.get_metadata()._convert_to_single_table()
+            if isinstance(synthesizer.metadata, Metadata):
+                metadata = synthesizer.metadata.tables[synthesizer._table_name]
+            else:
+                metadata = synthesizer.metadata
```

Comment:
what's wrong with keeping the convert method here?

Reply:
Backwards compatibility. When loading a synthesizer from a previous version, the stored metadata will be `SingleTableMetadata`. Because we updated the `get_metadata` method in this PR to simply return the stored metadata, `synthesizer.get_metadata()` will already be a `SingleTableMetadata` object.

Reply:
Ah ok. I think it would be a little less mysterious to put that in the `get_metadata` method instead of here. So in that method we can convert it to `Metadata` if it's an old one.